framework/include/mesh/FaceInfo.h

```diff
@@ -68,6 +68,10 @@ class FaceInfo
   /// Returns the coordinates of the face centroid.
   const Point & faceCentroid() const { return _face_centroid; }
 
+  /// Returns the coordinates of the approximate face centroid
+  /// in case of skewed meshes.
+  Point rInt() const { return _r_int; }
```

Comment:
what does r and Int stand for here?
Just trying to see if we can get a better name

Reply:
The position vector (r) for the intersection (int) of the face and the line connecting the cell centroids.

Reply:
rCFf lol

Reply:
faceIntersection()?

Reply:
Modified it to rIntersection() together with the corresponding member variable.

Reply:
Randomly jumping in here from an e-mail... why don't we return a reference here?

Reply:
It's probably safe to do so here. Any use of the return value of `rIntersection()` is likely to have a shorter lifetime than the `FaceInfo` object

Reply:
I get compiler warning if I do so: `returning reference to local temporary object [-Wreturn-stack-address]`

Reply:
with
``
const Point & rInt() const { return _r_int; }
``
?

Reply:
`_r_intersection` is a `RealVectorValue`. So you are constructing a temporary object (a `Point`) and then returning it. You either have to make `_r_intersection` a `Point` or return a reference to `RealVectorValue`

Reply:
Yes, exactly.

Reply:
👍🏼 just checking; like I said, I had no context here. Just saw the diff in an e-mail

Reply:
Ahhh, I see. I'll fix that in a sec.